allennlp/nn/samplers/samplers.py

```diff
@@ -17,22 +17,26 @@ def __init__(self, temperature: float = 1.0, filter_val: float = -float("inf"))
         self.filter_val = min_value_of_dtype(torch.float)
 
     def __call__(
-        self, logits: torch.Tensor, num_samples: int = 1, with_replacement: bool = True
+        self,
+        log_probs: torch.Tensor,
+        perturbed_log_probs: torch.Tensor = None,
+        num_samples: int = 1,
+        with_replacement: bool = True,
     ) -> torch.Tensor:
-        probabilities = torch.nn.functional.softmax(logits, dim=-1)
+        probabilities = torch.nn.functional.softmax(log_probs, dim=-1)
```

Comment:
`log_probs` are already normalized, so we can just do `log_probs.exp()` to get `probabilities`.